sdk/core/azure-core/azure/core/tracing/_inference_api_injector.py

```diff
@@ -0,0 +1,370 @@
+from azure.core.tracing import SpanKind
+from azure.core.settings import settings
+from .common import get_function_and_class_name
+
+_inference_traces_enabled: bool = False
+_trace_inference_content: bool = False
+
+class TraceType(str, Enum):
+    """An enumeration class to represent different types of traces."""
+
+    INFERENCE = "Inference"
+
+
+def _set_attributes(span: AbstractSpan, *attrs: tuple[str, Any]) -> None:
+    for attr in attrs:
+        key, value = attr
+        if value is not None:
+            span.add_attribute(key, value)
+
+
+def _add_request_chat_message_event(span: AbstractSpan, **kwargs: Any) -> None:
+    for message in kwargs.get("messages", []):
+        try:
+            message = message.as_dict()
+        except AttributeError:
+            pass
+
+        if message.get("role"):
+            name = f"gen_ai.{message.get('role')}.message"
+            span.span_instance.add_event(
+                name=name,
+                attributes={
+                    "get_ai.system": "openai",
+                    "gen_ai.event.content": json.dumps(message)
+                }
+            )
+
+
+def _add_request_chat_attributes(span: AbstractSpan, **kwargs: Any) -> None:
+    _set_attributes(
+        span,
+        ("gen_ai.system", "openai"),
+        ("gen_ai.request.model", kwargs.get("model")),
+        ("gen_ai.request.max_tokens", kwargs.get("max_tokens")),
+        ("gen_ai.request.temperature", kwargs.get("temperature")),
+        ("gen_ai.request.top_p", kwargs.get("top_p")),
+    )
+
+
+def _add_response_chat_message_event(span: AbstractSpan, result: _models.ChatCompletions) -> None:
+    global _trace_inference_content
+    for choice in result.choices:
+        if _trace_inference_content:
+            response: dict[str, Any] = {
+                "message": {"content": choice.message.content},
+                "finish_reason": str(choice.finish_reason),
+                "index": choice.index,
+            }
+            attributes={
+                "get_ai.system": "openai",
+                "gen_ai.event.content": json.dumps(response)
+            }
+        else:
+            response: dict[str, Any] = {
+                "finish_reason": str(choice.finish_reason),
+                "index": choice.index,
+            }
+            attributes={
+                "get_ai.system": "openai",
+            }
+        if choice.message.tool_calls:
+            response["message"]["tool_calls"] = [tool.as_dict() for tool in choice.message.tool_calls]
+        span.span_instance.add_event(name="gen_ai.choice", attributes=attributes)
+
+
+def _add_response_chat_attributes(span: AbstractSpan, result: _models.ChatCompletions | _models.StreamingChatCompletionsUpdate) -> None:
+    _set_attributes(
+        span,
+        ("gen_ai.response.id", result.id),
+        ("gen_ai.response.model", result.model),
+        ("gen_ai.response.finish_reason", str(result.choices[-1].finish_reason)),
```

Comment:
Are you assuming here that the choices array could have multiple cells? That's not the case for the common API (it does not support more than one element). So I would use `choices[0]` here instead of `choices[-1]`.